Sources/DataStructures/BinaryHeap.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  BinaryHeap.swift
+//  PitchSpeller
+//
+//  Created by Benjamin Wetherfield on 7/15/18.
+//
+
+/// Implements a priority queue where minimum values have highest priority
+struct BinaryHeap<Element: Hashable, Value: Comparable> {
```

Comment:
One last thing, can you add `Collection` conformance (you can do this just by forwarding a chain of `storage`, `lookup`, `indices` to `CollectionWrapping`):

```Swift
extension BinaryHeap: CollectionWrapping {
    var base: [Element] {
        ...
    }
}
```

Reply:
Hold on, I'm not sure about this... Does this not break the interface of the heap structure? I could be missing something in the organization of the project. The `CollectionWrapping` is definitely a nice refactor!

Reply:
Ah yes. You are totally right! Scratch this.